sdk/python/kfp/dsl/artifact.py

```diff
@@ -98,129 +46,105 @@ class Artifact(object):
      their components.
   2. At runtime, Artifact objects provide helper function/utilities to access
      the underlying RuntimeArtifact pb message, and provide additional layers
-     of validation to ensure type compatibility.
+     of validation to ensure type compatibility for fields specified in the 
+     instance schema.
   """
 
-  # Name of the Artifact type.
-  TYPE_NAME = 'Artifact'
-  # Property schema.
-  # Example usage:
-  #
-  # PROPERTIES = {
-  #   'span': Property(type=PropertyType.INT),
-  #   # Comma separated of splits for an artifact. Empty string means artifact
-  #   # has no split.
-  #   'split_names': Property(type=PropertyType.STRING),
-  # }
-  PROPERTIES = None
+  TYPE_NAME="Artifact"
+
+  # Dict holding the instance's schema fields used for type checking
+  _metadata_fields = Dict[str, MetadataFieldType]
```

Comment:
This is defining a type alias not a dict object. I think you meant to define an instance variable?

Reply:
This is similar to PROPERTIES  in the existing Artifact. As this is set in the init it is declared here.

Reply:
This specific syntax doesn't do what you want though. It's defining a type alias at the class level, rather than a variable declaration which should be `_metadata_fields: Dict[str, MetadataFiledType] = {}`.
Besides as a class variable, this is shadowed by the instance variable you defined in __init__. So you don't actually need this line here.

Reply:
Oh I see what you mean. Makes sense.